### ISSUE TITLE
New rule: Local variables should not shadow class fields

### ISSUE THREAD

**Issue by meziantou** [meziantou/Meziantou.Analyzer#76]
It avoid problems such as

````c#
public class Sample
{
    private string field;
    public Sample(string field)
    {
        field = field; // Assign parameter instead of field
    }
}
````

BTW, it should also detect assignment of the same variable

````
var a = 10;
a = a; // Error
````

**Comment by meziantou** [comment]
Fixed by 40ebcab9a09c7b3b358506e2cced92965be677a5
